client/src/context/ProfileContext.js

```diff
@@ -0,0 +1,168 @@
+import React, { useReducer } from 'react';
+import axios from 'axios';
+
+
+//Created Context
+export const ProfileContext = React.createContext();
+
+
+//Stored action types names
+
+const types = {
+	Add_Profile:"Add_Profile",
+	Get_Profile: "Get_Profile",
+	Get_Profiles: "Get_Profiles",
+	Delete_Profile: "Delete_Profile",
+	Edit_Profile: "Edit_Profile",
+	Profile_Error: "Profile_Error",
+	Clear_Profile :"Clear_Profile",
+	Open_Edit :"Open_Edit",
+};
+
+
+//Stored actions in a reducer
+
+const profileReducer = (state, action) => {
+	switch (action.type) {
+	case types.Profile_Error:
+		return { ...state, errorMessage: action.payload, loading: false };
+	case types.Add_Profile:
+		return { ...state, profiles: [...state.profiles, action.payload], loading: false };
+	case types.Get_Profile:
+		return { ...state, profile: action.payload, loading: false };
+	case types.Clear_Profile:
+		return { ...state, profile: null, loading: false };
+	case types.Get_Profiles:
+		return { ...state, profiles: action.payload, loading: false };
+	case types.Edit_Profile:
+		return { ...state, profile: action.payload, loading: false };
+	case types.Delete_Applicaton:
+		return { ...state, profiles: state.profiles.filter((profile) => profile.id !== action.payload), loading: false };
+	case types.Open_Edit:
+		return { ...state, edit:action.payload };
+	default:
+		return state;
+	}
+};
+
+
+//Context has Provider and Consumer in itself.
+//Here we set Context Provider and stored our states and actions in it to provide them to Consumer
+
+const ProfileState = (props) =>{
+
+	const initialState ={
+		profiles:[],
+		profile:null,
+		edit:false,
+	};
+
+	const [state, dispatch] = useReducer(profileReducer, initialState);
+
+	const getProfiles = async ()=>{
+		const response = await axios.get(`https://jsonplaceholder.typicode.com/users`);
```

Comment:
It might be nice to extract the api calls to it's own API.js file and import, something like `fetchProfiles()` would make sense to me. `fetch*` is quite a common pattern to sign it's going to be loading something from an external source from my experience.